src/unity/python/turicreate/test/test_one_shot_object_detector.py

```diff
@@ -0,0 +1,233 @@
+        img = rs.randint(255, size=img_shape)
+
+        pil_img = _PIL_Image.fromarray(img, mode='RGB')
+        # Randomly select image format
+        image_format = FORMATS[rs.randint(len(FORMATS))]
+        images.append(from_pil_image(pil_img, image_format=image_format))
+
+        
+
+    starter_images = []
+    starter_target = []
+    for i in range(num_starter_images):
+        img_shape = tuple(rs.randint(100, 1000, size=2)) + (3,)
+        img = rs.randint(255, size=img_shape)
+        pil_img = _PIL_Image.fromarray(img, mode='RGB')
+        image_format = FORMATS[rs.randint(len(FORMATS))]
+        starter_images.append(from_pil_image(pil_img, image_format=image_format))
+        starter_target.append(i % len(_CLASSES))
+
+    train = tc.SFrame({
+        feature: tc.SArray(starter_images),
+        target: tc.SArray(starter_target),
+    })
+    test = tc.SFrame({
+        feature: tc.SArray(images),
+    })
+    backgrounds = test[feature].head(5)
+    return train, test, backgrounds
+
+
+class OneObjectDetectorSmokeTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        """
+        The setup class method for the basic test case with all default values.
+        """
+        self.feature = 'myimage'
+        self.target = 'mytarget'
+
+        ## Create the model
+        self.def_opts = {
+           'model': 'darknet-yolo',
+           'max_iterations': 2,
+        }
+
+        # Model
+        self.train, self.test, self.backgrounds = _get_data(feature=self.feature, target=self.target)
+        self.model = tc.one_shot_object_detector.create(self.train,
+                                                        target=self.target,
+                                                        backgrounds=self.backgrounds,
+                                                        batch_size=2,
+                                                        max_iterations=1)
+
+        ## Answers
+        self.opts = self.def_opts.copy()
+        self.opts['max_iterations'] = 1
+
+        self.get_ans = {
+           'target': lambda x: x == self.target,
+           'num_starter_images': lambda x: x > 0,
+           'num_classes': lambda x: x == len(_CLASSES),
+           'detector': lambda x: isinstance(x, tc.object_detector.object_detector.ObjectDetector),
+           '_detector_version': lambda x: x==1
+        }
+        self.fields_ans = self.get_ans.keys()
+
+    def test_create_with_missing_target(self):
+        with self.assertRaises(_ToolkitError):
+            tc.one_shot_object_detector.create(self.train, target='wrong_feature', 
+                backgrounds=self.backgrounds, max_iterations=1)
+
+    def test_create_with_empty_dataset(self):
+        with self.assertRaises(_ToolkitError):
+            tc.one_shot_object_detector.create(self.train[:0], target=self.target)
+
+    def test_predict(self):
+        sf = self.test.head()
+        # Make sure this does not need the annotations column to work
```

Comment:
I don't understand this comment. The object detector's prediction function doesn't even take an annotation column.

Reply:
Yeah I deleted the code but forgot to delete the comment.